### ISSUE TITLE
In Docker provider 'networks' appends new nets to the default bridge

### ISSUE THREAD

**Issue by fdebonneval** [hashicorp/terraform#12095]
Hi there, I hope this issue have not already been reported, I took a look and did not find duplicate.
Anyway, thanks for telling me if it is an expected behavior or a genuine issue to fix.

### Terraform Version
`# terraform --version
Terraform v0.8.7` 

### Affected Resource(s)
- docker networks

### Terraform Configuration Files
```
provider "docker" {
    host = "unix:///var/run/docker.sock"
}

resource "docker_container" "alpine" {
  name = "foo"
  image = "alpine:latest"
  command = ["ip", "a"]
  networks = ["tf-net"]
}
```

### Debug Output
The `docker logs` of the container returns
```
[...]
44: eth0@if45: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1500 qdisc noqueue state UP
    link/ether 02:42:ac:11:00:03 brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.3/16 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::42:acff:fe11:3/64 scope link tentative
       valid_lft forever preferred_lft forever
46: eth1@if47: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1500 qdisc noqueue state UP
    link/ether 02:42:ac:12:00:02 brd ff:ff:ff:ff:ff:ff
    inet 172.18.0.2/16 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::42:acff:fe12:2/64 scope link tentative
       valid_lft forever preferred_lft fore
```

Also `Docker inspect` of the created container : 
```
[...]
"Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "8eac1874812a79759249b97e8aba68f20256e5f772a5bdcc033271d0120b86a4",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": ""
                },
                "tf-net": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "9c3bf9a9296d"
                    ],
                    "NetworkID": "71ce9a1e51eaa9a6098881c946f5099bccebe2728158f44ce374213ee9041cb8",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": ""
                }
            }
[...]
```

### Expected Behavior
With only one network in the list, I would expect only one 'eth0' interface inside the running container 

### Actual Behavior
I get the default bridge and the network from the list.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. On a docker host
2. Define a .tf file from the example above
3. The run an apply `docker run --rm -it -v /var/run/docker.sock:/var/run/docker.sock -v $(pwd):/tf --workdir /tf hashicorp/terraform:light apply`

**Comment by sebastien-prudhomme** [comment]
Same behavior with Docker 1.13.1 and Terraform 0.9.0-beta1. The container is always attached to the default bridge network.

**Comment by sebastien-prudhomme** [comment]
Don't know if it can be considered as a workaround: use 'network_mode' option for your main network and 'networks' option only for supplementary networks.

```
data "docker_registry_image" "httpd" {
  name = "httpd:latest"
}

resource "docker_image" "httpd" {
  name          = "${data.docker_registry_image.httpd.name}"
  keep_locally  = "true"
  pull_triggers = ["${data.docker_registry_image.httpd.sha256_digest}"]
}

resource "docker_network" "httpd" {
  name = "httpd"
}

resource "docker_container" "httpd" {
  name     = "httpd"
  image    = "${docker_image.httpd.latest}"
  hostname = "httpd"
  restart  = "unless-stopped"

  ports {
    internal = "80"
    external = "80"
  }

  network_mode = "${docker_network.httpd.name}"
}
```
This results in only one ethernet interface:

```
docker exec httpd ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
977: eth0@if978: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether 02:42:ac:17:00:02 brd ff:ff:ff:ff:ff:ff
    inet 172.23.0.2/16 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::42:acff:fe17:2/64 scope link 
       valid_lft forever preferred_lft forever
```

**Comment by fdebonneval** [comment]
Hi Sebastien,
Thanks for the advice. I do not really see how the network mode changes the behavior ... I should give it a try.
From my side as I wrapped the commands in a Makefile, I just added a `docker network disconnect` after my terraform apply.
I also tried to unplug the network with a local-exec, but the terraform light image does not carries the docker binary.

Anyway thanks for the advice.
Foucault
